.travis.yml

```diff
@@ -47,6 +47,35 @@ matrix:
            PROJECT_URL=https://github.com/joblib/joblib.git
       python: 3.7
       if: commit_message =~ /(\[ci downstream\]|\[ci joblib\])/
+    - os: linux
+      env: PROJECT=ray TEST_REQUIREMENTS="setproctitle hypothesis psutil"
+           PROJECT_URL=https://github.com/ray-project/ray.git
+           PYTEST_ARGS="--duration=10 -k test_mini or test_basic"
+      python: 3.7
+      if: commit_message =~ /(\[ci downstream\]|\[ci ray\])/
+      install:
+        - pip install .
+        - pip install --upgrade -r dev-requirements.txt
+        - pip install tornado
+        - if [[ $TRAVIS_PYTHON_VERSION != 'pypy'* ]]; then
+              pip install numpy scipy;
+          fi
+        - pip install $TEST_REQUIREMENTS;
+        - pip install ray==0.6.4;
+        - PROJECT_DIR="$(python -c "import os,ray,sys;print(os.path.dirname(ray.__file__));sys.stdout.flush()")";
+        - rm $PROJECT_DIR/cloudpickle/cloudpickle.py;
+        - cp cloudpickle/cloudpickle.py $PROJECT_DIR/cloudpickle/cloudpickle.py;
+        - PROJECT_DIR="$PROJECT_DIR/tests";
+        - pip list
+      script:
+        - COVERAGE_PROCESS_START="$TRAVIS_BUILD_DIR/.coveragerc" PYTHONPATH='.:tests' pytest -r s
+        - pushd $PROJECT_DIR
+        - pytest -vl --duration=10 $PROJECT_DIR/test_mini.py $PROJECT_DIR/test_basic.py
```

Comment:
Let's start with just `test_mini.py` and remove `test_basic.py`. `test_basic.py` takes a while to run and there are one or two flaky tests.